src/modules/private-share-folder/private-sharing.usecase.ts

```diff
@@ -2,12 +2,42 @@ import { Injectable } from '@nestjs/common';
 import { Folder } from '../folder/folder.domain';
 import { User } from '../user/user.domain';
 import { SequelizePrivateSharingRepository } from './private-sharing.repository';
+import { SequelizeUserRepository } from '../user/user.repository';
+import { SequelizeFolderRepository } from '../folder/folder.repository';
 
 @Injectable()
 export class PrivateSharingUseCase {
   constructor(
     private privateSharingRespository: SequelizePrivateSharingRepository,
+    private userRespository: SequelizeUserRepository,
+    private folderRespository: SequelizeFolderRepository,
   ) {}
+  async grantPrivileges(
+    owner: User,
+    userUuid: string,
+    privateFolderId: string,
+    roleUuid: string,
+  ) {
+    const user = await this.userRespository.findByUuid(userUuid);
+    const privateFolder = await this.privateSharingRespository.findById(
+      privateFolderId,
+    );
+    const folder = await this.folderRespository.findByUuid(
+      privateFolder.folderId,
+    );
+
+    const isOwner = await this.folderRespository.isOwner(owner, folder.id);
```

Comment:
There are unnecessary queries here. The user cames already loaded by the middleware, which by the way, already does query the user against the database. 

Due to that, I recommend to make it still secure but faster and easier to just query the folder by user id or just better, to ensure that when we finally migrate to user uuids definitely: 

Query the folder by the uuid and if it is the owner, just check that `user.id === folder.userId`

That requires just 2 queries, easier to maintain, to update and to test.

Reply:
You will still have the user and folder objects required to create the private folder role of the user inside that folder